h2/connection.py

```diff
@@ -1922,6 +1927,23 @@ def _receive_alt_svc_frame(self, frame):
 
         return frames, events
 
+    def _receive_unknown_frame(self, frame):
+        """
+        We have received a frame that we do not understand. This is almost
+        certainly an extension frame, though it's impossible to be entirely
+        sure.
+
+        RFC 7540 § 5.5 says that we MUST ignore unknown frame types: so we
+        do. We do notify the user that we received one, however.
+        """
+        # All we do here is log.
+        self.config.logger.debug(
+            "Received unknown extension frame (ID 0x%x)", frame.stream_id
```

Comment:
FWIW, I would probably lean towards doing logging in hex as you did it originally, but let’s make that a separate patch.